In [1]:
import pandas as pd

In [2]:
data2016 = pd.read_csv('C:\\Users\\blee\\Springboard\\kaggle2016.csv', encoding = 'latin1')

C:\Users\blee\Anaconda3\lib\site-packages\IPython\core\interactiveshell.py:3146: DtypeWarning: Columns (13,14,15) have mixed types.Specify dtype option on import or set low_memory=False.
  has_raised = await self.run_ast_nodes(code_ast.body, cell_name,


In [3]:
data2016.head()

ID                                               name   \
0  1000002330                    The Songs of Adelaide & Abullah   
1  1000004038                                     Where is Hank?   
2  1000007540  ToshiCapital Rekordz Needs Help to Complete Album   
3  1000011046  Community Film Project: The Art of Neighborhoo...   
4  1000014025                               Monarch Espresso Bar   

        category  main_category  currency         deadline   goal   \
0          Poetry     Publishing       GBP  10/9/2015 11:36   1000   
1  Narrative Film   Film & Video       USD   2/26/2013 0:20  45000   
2           Music          Music       USD   4/16/2012 4:24   5000   
3    Film & Video   Film & Video       USD   8/29/2015 1:00  19500   
4     Restaurants           Food       USD   4/1/2016 13:38  50000   

         launched  pledged       state  backers  country  usd pledged   \
0  8/11/2015 12:12        0      failed        0       GB            0   
1   1/12/2013 0:20      220      failed        3       US          220   
2   3/17/2012 3:24        1      failed        1       US            1   
3    7/4/2015 8:35     1283    canceled       14       US         1283   
4  2/26/2016 13:38    52375  successful      224       US        52375   

  Unnamed: 13 Unnamed: 14 Unnamed: 15  Unnamed: 16  
0         NaN         NaN         NaN          NaN  
1         NaN         NaN         NaN          NaN  
2         NaN         NaN         NaN          NaN  
3         NaN         NaN         NaN          NaN  
4         NaN         NaN         NaN          NaN

In [4]:
data2016.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 323750 entries, 0 to 323749
Data columns (total 17 columns):
 #   Column          Non-Null Count   Dtype  
---  ------          --------------   -----  
 0   ID              323750 non-null  int64  
 1   name            323746 non-null  object 
 2   category        323745 non-null  object 
 3   main_category   323750 non-null  object 
 4   currency        323750 non-null  object 
 5   deadline        323750 non-null  object 
 6   goal            323750 non-null  object 
 7   launched        323750 non-null  object 
 8   pledged         323750 non-null  object 
 9   state           323750 non-null  object 
 10  backers         323750 non-null  object 
 11  country         323750 non-null  object 
 12  usd pledged     319960 non-null  object 
 13  Unnamed: 13     625 non-null     object 
 14  Unnamed: 14     12 non-null      object 
 15  Unnamed: 15     4 non-null       object 
 16  Unnamed: 16     1 non-null       float64
dtypes: float64

In [5]:
unnamed13_notnull = data2016['Unnamed: 13'].notnull() == True

In [6]:
unnamed13_notnull_dataframe = data2016[unnamed13_notnull]
unnamed13_notnull_dataframe

ID                                      name   \
1454    1008705746                 Zephyra´s new full length   
1563    1009317190                            French Cuisine   
1794    1010871699   The Beginners Guide to being Unsuicidal   
1931    1011687764                   Best OnLine Classifieds   
2420     101453314               Social Media Ruined My Life   
...            ...                                       ...   
321945   989007729                           THROUGH MY EYES   
322162   990511774  Daniel Hresko's new CD is (almost) ready   
322204   990746749                        Feet on the Ground   
323138   996542939         'WANDER' - Apocalyptic Short Film   
323495    99851211           Art/sustainability/spirituality   

                                category   main_category      currency   \
1454             'As The World Collapses'           Metal         Music   
1563             A Traditional Experience       Cookbooks          Food   
1794                          the one act         Theater       Theater   
1931                  Ever / No More Spam             Web    Technology   
2420      A Short Film from Adam S Curtis          Shorts  Film & Video   
...                                   ...             ...           ...   
321945   MY LIFE IN THE MISSISSIPPI DELTA  Narrative Film  Film & Video   
322162                   so get on board!      Indie Rock         Music   
322204                 Head in the Clouds    Film & Video  Film & Video   
323138                    Post Production          Shorts  Film & Video   
323495        a bike quest in New Zealand     Photography   Photography   

       deadline              goal  launched           pledged  state   \
1454         SEK     2/2/2016 0:56     15000     1/3/2016 0:56   4262   
1563         USD     9/8/2014 0:46     13730     8/9/2014 3:16   3984   
1794         USD  12/31/2011 23:25      5000  11/21/2011 23:25    525   
1931         USD   9/20/2014 19:56      6300   8/21/2014 19:56      0   
2420         USD   3/14/2013 20:11      3000   2/25/2013 21:11   3035   
...          ...               ...       ...               ...    ...   
321945       USD     4/5/2012 2:34      3500     3/6/2012 2:34     10   
322162       USD     9/6/2011 5:59       400    8/6/2011 15:36     61   
322204       USD   4/17/2013 16:00     35000   3/18/2013 15:00    179   
323138       GBP   4/26/2015 17:12      1500   2/25/2015 17:12   2501   
323495       USD    3/5/2013 21:00      2500     2/5/2013 8:09   2634   

          backers  country  usd pledged   Unnamed: 13 Unnamed: 14 Unnamed: 15  \
1454        failed       14           SE  504.9476528         NaN         NaN   
1563        failed       46           US         3984         NaN         NaN   
1794        failed       10           US          525         NaN         NaN   
1931        failed        0           US            0         NaN         NaN   
2420    successful       42           US         3035         NaN         NaN   
...            ...      ...          ...          ...         ...         ...   
321945      failed        1           US           10         NaN         NaN   
322162      failed        4           US           61         NaN         NaN   
322204      failed       10           US          179         NaN         NaN   
323138  successful       50           GB   3861.84447         NaN         NaN   
323495  successful       34           US         2634         NaN         NaN   

        Unnamed: 16  
1454            NaN  
1563            NaN  
1794            NaN  
1931            NaN  
2420            NaN  
...             ...  
321945          NaN  
322162          NaN  
322204          NaN  
323138          NaN  
323495          NaN  

[625 rows x 17 columns]

In [7]:
data2016.loc[[1454]]

ID                       name                   category   \
1454  1008705746  Zephyra´s new full length   'As The World Collapses'   

     main_category  currency  deadline           goal  launched   \
1454          Metal     Music       SEK  2/2/2016 0:56     15000   

           pledged  state  backers  country  usd pledged   Unnamed: 13  \
1454  1/3/2016 0:56   4262   failed       14           SE  504.9476528   

     Unnamed: 14 Unnamed: 15  Unnamed: 16  
1454         NaN         NaN          NaN

Was trying to confirm if data in "Unnamed: 13" is the US currency conversion as posted from the datasource, but instead, we find that the data columns are misaligned. Somewhere between "category" and "main_category" the rows seem to extend into the "currency" column and shift all the data 1 column over. Since there are 625 rows like this, it doesn't make sense to drop them all, and instead, we need to figure out a way to shift the columns back over. 

These are all shifted over by 1 column. It looks like their name spilled into the "category" column. We could delete what is in the "category" column now, and then shift everything over. 

It looks like all the not null columns in "Unnamed: 14" are also not null for "Unnamed: 13." It also appears that those not null in "Unnamed: 14" are also shifted data, where the column names do not align with the actual data stored in that cell. The 'country' column should not have inputs of "failed", "successful", or numbers and dates. 

In [8]:
unnamed16= data2016[data2016['Unnamed: 16'].notnull() == True]
unnamed16

ID          name  category  main_category    currency   \
269970  677103185  SixSixSeven    Angels         Demons    Religion    

        deadline            goal  launched  pledged            state   \
269970   Esoteric  Graphic Novels    Comics      USD  10/10/2015 1:00   

       backers          country  usd pledged  Unnamed: 13 Unnamed: 14  \
269970      750  9/10/2015 18:15           25      failed           1   

       Unnamed: 15  Unnamed: 16  
269970          US         25.0

the same is true for unnamed 15 and unnamed 16, both of these column only have not null values for rows that seem to have shifted data. It looks like the rows with multiple category or main_category entries, seperated by a comma, when uploaded, is what caused the data to shift. 

In [9]:
unnamed16 = data2016.iloc[[269970]]
unnamed16

ID          name  category  main_category    currency   \
269970  677103185  SixSixSeven    Angels         Demons    Religion    

        deadline            goal  launched  pledged            state   \
269970   Esoteric  Graphic Novels    Comics      USD  10/10/2015 1:00   

       backers          country  usd pledged  Unnamed: 13 Unnamed: 14  \
269970      750  9/10/2015 18:15           25      failed           1   

       Unnamed: 15  Unnamed: 16  
269970          US         25.0

take columns shifted, concat to new column, delete old columns. For remainder data, do column rename. then append back into main data

In [10]:
unnamed16.columns

Index(['ID ', 'name ', 'category ', 'main_category ', 'currency ', 'deadline ',
       'goal ', 'launched ', 'pledged ', 'state ', 'backers ', 'country ',
       'usd pledged ', 'Unnamed: 13', 'Unnamed: 14', 'Unnamed: 15',
       'Unnamed: 16'],
      dtype='object')

Notice that the column names have a space after the letters

In [11]:
unnamed16['name '] = unnamed16['name '] + unnamed16['category '] + unnamed16['main_category '] + unnamed16['currency '] + unnamed16['deadline ']

<ipython-input-11-1aa14930915f>:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  unnamed16['name '] = unnamed16['name '] + unnamed16['category '] + unnamed16['main_category '] + unnamed16['currency '] + unnamed16['deadline ']


creating a new separate dataframe so that we can merge this back with the original data, and delete the original misaligned column/values so we can ignore this warning 

In [12]:
unnamed16

ID                                             name  category   \
269970  677103185  SixSixSeven  Angels  Demons  Religion  Esoteric   Angels    

       main_category    currency   deadline            goal  launched   \
269970        Demons    Religion    Esoteric  Graphic Novels    Comics   

       pledged            state  backers          country  usd pledged   \
269970      USD  10/10/2015 1:00      750  9/10/2015 18:15           25   

       Unnamed: 13 Unnamed: 14 Unnamed: 15  Unnamed: 16  
269970      failed           1          US         25.0

In [13]:
unnamed16.drop(['category ', 'main_category ', 'currency ', 'deadline '], axis =1, inplace = True)


C:\Users\blee\Anaconda3\lib\site-packages\pandas\core\frame.py:4163: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  return super().drop(


In [14]:
column_names = ['ID', 'name', 'category', 'main_category', 'currency', 'deadline', 'goal', 'launched', 'pledged', 'state', 'backers', 'country', 'usd_pledged']

In [15]:
unnamed16.columns = column_names

In [16]:
unnamed16

ID                                             name  \
269970  677103185  SixSixSeven  Angels  Demons  Religion  Esoteric   

              category main_category currency         deadline goal  \
269970  Graphic Novels        Comics      USD  10/10/2015 1:00  750   

               launched pledged   state backers country  usd_pledged  
269970  9/10/2015 18:15      25  failed       1      US         25.0

this method worked for 1 row, now let's try it on the unnamed 15, 14, and 13 options 

In [17]:
data2016.drop(269970, inplace = True)

Dropping this row from the main data2016 dataframe will prevent this row from showing up in the unnamed15 list, since the orignal dataframe will still have the row 269970 as a non null value for the columns in "Unnamed: 15". After clean the data, so that all only have 13 columns, we will delete the last 3 columns from data2016, rename the columns to column_names, and then append the new dataframes (unnamed16, 15, 14, and 13)

In [18]:
unnamed15= data2016[data2016['Unnamed: 15'].notnull() == True]
unnamed15

ID                                name       category   \
104120  1618382802  Druid Hill Park Passport: Discover          Enjoy   
145502  1864561559                             T-shirt   dress shirts   
264243   642928449                  FRANCIS - The Pope   Buenos Aires   

       main_category            currency      deadline          goal   \
104120          Learn          Be active!    Publishing    Publishing   
145502          pants    clothing company       Apparel       Fashion   
264243         soccer   tango and culture  Film & Video  Film & Video   

       launched         pledged  state         backers  country  usd pledged   \
104120       USD  7/28/2012 1:30   9500  6/6/2012 23:54     9854   successful   
145502       USD  8/2/2015 18:23  20000  7/3/2015 18:23        0       failed   
264243       USD  3/2/2014 23:34  75000   2/5/2014 2:06        0       failed   

       Unnamed: 13 Unnamed: 14 Unnamed: 15  Unnamed: 16  
104120         208          US        9854          NaN  
145502           0          US           0          NaN  
264243           0          US           0          NaN

In [19]:
unnamed15.columns

Index(['ID ', 'name ', 'category ', 'main_category ', 'currency ', 'deadline ',
       'goal ', 'launched ', 'pledged ', 'state ', 'backers ', 'country ',
       'usd pledged ', 'Unnamed: 13', 'Unnamed: 14', 'Unnamed: 15',
       'Unnamed: 16'],
      dtype='object')

In [20]:
unnamed15['name '] = unnamed15['name '] + unnamed15['category '] + unnamed15['main_category '] + unnamed15['currency ']


<ipython-input-20-1e8559d8bfc2>:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  unnamed15['name '] = unnamed15['name '] + unnamed15['category '] + unnamed15['main_category '] + unnamed15['currency ']


In [21]:
unnamed15.drop(['category ', 'main_category ', 'currency ', 'Unnamed: 16'], inplace = True, axis = 1)


C:\Users\blee\Anaconda3\lib\site-packages\pandas\core\frame.py:4163: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  return super().drop(


In [22]:
unnamed15.columns = column_names
unnamed15

ID                                               name  \
104120  1618382802  Druid Hill Park Passport: Discover Enjoy Learn...   
145502  1864561559        T-shirt dress shirts pants clothing company   
264243   642928449  FRANCIS - The Pope Buenos Aires soccer tango a...   

            category main_category currency        deadline   goal  \
104120    Publishing    Publishing      USD  7/28/2012 1:30   9500   
145502       Apparel       Fashion      USD  8/2/2015 18:23  20000   
264243  Film & Video  Film & Video      USD  3/2/2014 23:34  75000   

              launched pledged       state backers country usd_pledged  
104120  6/6/2012 23:54    9854  successful     208      US        9854  
145502  7/3/2015 18:23       0      failed       0      US           0  
264243   2/5/2014 2:06       0      failed       0      US           0

In [23]:
data2016.drop([104120, 145502, 264243], inplace = True)

In [24]:
unnamed14 = data2016[data2016['Unnamed: 14'].notnull() == True]
unnamed14

ID                              name   \
13795   1081139420                The Rolling Stones   
16482   1097082409                         Pixel art   
61447   1365419849                               VIA   
112934  1670090635          EXCALIBRE:Sword of Truth   
225692   410977509   Handmade Artisan Soaps: Natural   
243349   516522480                    American Booty   
275186   708058934                           Be Wise   
305503   890269949  Community Clothing. Make Clothes   

                                 category            main_category   \
13795                      BEGGARS BANQUET   50 Years in the Making   
16482                         personalized        Retro Gaming art.   
61447    ME (V.isual I.nteractive A.rchive       M.ercury E.dition)   
112934                       Path of Light       Kingdom of Heaven.   
225692                              Gentle       Good for your skin   
243349        USA-made products Series One                  Spirits   
275186                         Wear Wisdom            Divine Wisdom   
305503                         Create Jobs            Restore Pride   

         currency    deadline  goal         launched  pledged   \
13795         Rock       Music   USD   8/12/2011 1:17     4625   
16482       Crafts      Crafts   USD  7/28/2016 21:36      500   
61447     Software  Technology   USD  3/19/2016 20:40    10000   
112934  Nonfiction  Publishing   USD    4/1/2016 8:00    24000   
225692      Crafts      Crafts   USD  10/8/2015 22:08      200   
243349    Painting         Art   USD    2/2/2013 3:45     1776   
275186     Apparel     Fashion   USD  2/27/2015 13:53    10000   
305503     Fashion     Fashion   GBP  3/16/2016 16:24    75000   

                 state  backers     country  usd pledged  Unnamed: 13  \
13795     6/4/2011 1:17       20      failed            2          US   
16482   6/28/2016 21:36        0      failed            0          US   
61447   2/18/2016 21:40       26      failed            2          US   
112934  2/22/2016 21:57        1      failed            1          US   
225692   9/8/2015 22:08      435  successful           10          US   
243349    1/3/2013 3:45      401      failed            7          US   
275186  1/28/2015 13:53        1      failed            1          US   
305503  2/15/2016 17:24  88619.2  successful         1020          GB   

       Unnamed: 14 Unnamed: 15  Unnamed: 16  
13795           20         NaN          NaN  
16482            0         NaN          NaN  
61447           26         NaN          NaN  
112934           1         NaN          NaN  
225692         435         NaN          NaN  
243349         401         NaN          NaN  
275186           1         NaN          NaN  
305503      128535         NaN          NaN

In [25]:
unnamed14['name '] = unnamed14['name '] + unnamed14['category '] + unnamed14['main_category ']
unnamed14.drop(['category ', 'main_category ', 'Unnamed: 16', 'Unnamed: 15'], inplace = True, axis = 1)


<ipython-input-25-2636e96a1ebf>:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  unnamed14['name '] = unnamed14['name '] + unnamed14['category '] + unnamed14['main_category ']
C:\Users\blee\Anaconda3\lib\site-packages\pandas\core\frame.py:4163: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  return super().drop(


In [26]:
unnamed14.columns = column_names
unnamed14

ID                                               name  \
13795   1081139420  The Rolling Stones BEGGARS BANQUET 50 Years in...   
16482   1097082409           Pixel art personalized Retro Gaming art.   
61447   1365419849  VIA ME (V.isual I.nteractive A.rchive M.ercury...   
112934  1670090635  EXCALIBRE:Sword of Truth Path of Light Kingdom...   
225692   410977509  Handmade Artisan Soaps: Natural Gentle Good fo...   
243349   516522480  American Booty USA-made products Series One Sp...   
275186   708058934                  Be Wise Wear Wisdom Divine Wisdom   
305503   890269949  Community Clothing. Make Clothes Create Jobs R...   

          category main_category currency         deadline   goal  \
13795         Rock         Music      USD   8/12/2011 1:17   4625   
16482       Crafts        Crafts      USD  7/28/2016 21:36    500   
61447     Software    Technology      USD  3/19/2016 20:40  10000   
112934  Nonfiction    Publishing      USD    4/1/2016 8:00  24000   
225692      Crafts        Crafts      USD  10/8/2015 22:08    200   
243349    Painting           Art      USD    2/2/2013 3:45   1776   
275186     Apparel       Fashion      USD  2/27/2015 13:53  10000   
305503     Fashion       Fashion      GBP  3/16/2016 16:24  75000   

               launched  pledged       state backers country usd_pledged  
13795     6/4/2011 1:17       20      failed       2      US          20  
16482   6/28/2016 21:36        0      failed       0      US           0  
61447   2/18/2016 21:40       26      failed       2      US          26  
112934  2/22/2016 21:57        1      failed       1      US           1  
225692   9/8/2015 22:08      435  successful      10      US         435  
243349    1/3/2013 3:45      401      failed       7      US         401  
275186  1/28/2015 13:53        1      failed       1      US           1  
305503  2/15/2016 17:24  88619.2  successful    1020      GB      128535

In [27]:
unnamed14.index

Int64Index([13795, 16482, 61447, 112934, 225692, 243349, 275186, 305503], dtype='int64')

In [28]:
data2016.drop(unnamed14.index, inplace = True)

In [29]:
data2016.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 323738 entries, 0 to 323749
Data columns (total 17 columns):
 #   Column          Non-Null Count   Dtype  
---  ------          --------------   -----  
 0   ID              323738 non-null  int64  
 1   name            323734 non-null  object 
 2   category        323733 non-null  object 
 3   main_category   323738 non-null  object 
 4   currency        323738 non-null  object 
 5   deadline        323738 non-null  object 
 6   goal            323738 non-null  object 
 7   launched        323738 non-null  object 
 8   pledged         323738 non-null  object 
 9   state           323738 non-null  object 
 10  backers         323738 non-null  object 
 11  country         323738 non-null  object 
 12  usd pledged     319948 non-null  object 
 13  Unnamed: 13     613 non-null     object 
 14  Unnamed: 14     0 non-null       object 
 15  Unnamed: 15     0 non-null       object 
 16  Unnamed: 16     0 non-null       float64
dtypes: float64

In [30]:
unnamed13 = data2016[data2016['Unnamed: 13'].notnull() == True]
unnamed13

ID                                      name   \
1454    1008705746                 Zephyra´s new full length   
1563    1009317190                            French Cuisine   
1794    1010871699   The Beginners Guide to being Unsuicidal   
1931    1011687764                   Best OnLine Classifieds   
2420     101453314               Social Media Ruined My Life   
...            ...                                       ...   
321945   989007729                           THROUGH MY EYES   
322162   990511774  Daniel Hresko's new CD is (almost) ready   
322204   990746749                        Feet on the Ground   
323138   996542939         'WANDER' - Apocalyptic Short Film   
323495    99851211           Art/sustainability/spirituality   

                                category   main_category      currency   \
1454             'As The World Collapses'           Metal         Music   
1563             A Traditional Experience       Cookbooks          Food   
1794                          the one act         Theater       Theater   
1931                  Ever / No More Spam             Web    Technology   
2420      A Short Film from Adam S Curtis          Shorts  Film & Video   
...                                   ...             ...           ...   
321945   MY LIFE IN THE MISSISSIPPI DELTA  Narrative Film  Film & Video   
322162                   so get on board!      Indie Rock         Music   
322204                 Head in the Clouds    Film & Video  Film & Video   
323138                    Post Production          Shorts  Film & Video   
323495        a bike quest in New Zealand     Photography   Photography   

       deadline              goal  launched           pledged  state   \
1454         SEK     2/2/2016 0:56     15000     1/3/2016 0:56   4262   
1563         USD     9/8/2014 0:46     13730     8/9/2014 3:16   3984   
1794         USD  12/31/2011 23:25      5000  11/21/2011 23:25    525   
1931         USD   9/20/2014 19:56      6300   8/21/2014 19:56      0   
2420         USD   3/14/2013 20:11      3000   2/25/2013 21:11   3035   
...          ...               ...       ...               ...    ...   
321945       USD     4/5/2012 2:34      3500     3/6/2012 2:34     10   
322162       USD     9/6/2011 5:59       400    8/6/2011 15:36     61   
322204       USD   4/17/2013 16:00     35000   3/18/2013 15:00    179   
323138       GBP   4/26/2015 17:12      1500   2/25/2015 17:12   2501   
323495       USD    3/5/2013 21:00      2500     2/5/2013 8:09   2634   

          backers  country  usd pledged   Unnamed: 13 Unnamed: 14 Unnamed: 15  \
1454        failed       14           SE  504.9476528         NaN         NaN   
1563        failed       46           US         3984         NaN         NaN   
1794        failed       10           US          525         NaN         NaN   
1931        failed        0           US            0         NaN         NaN   
2420    successful       42           US         3035         NaN         NaN   
...            ...      ...          ...          ...         ...         ...   
321945      failed        1           US           10         NaN         NaN   
322162      failed        4           US           61         NaN         NaN   
322204      failed       10           US          179         NaN         NaN   
323138  successful       50           GB   3861.84447         NaN         NaN   
323495  successful       34           US         2634         NaN         NaN   

        Unnamed: 16  
1454            NaN  
1563            NaN  
1794            NaN  
1931            NaN  
2420            NaN  
...             ...  
321945          NaN  
322162          NaN  
322204          NaN  
323138          NaN  
323495          NaN  

[613 rows x 17 columns]

In [31]:
unnamed13['name '] = unnamed13['name '] + unnamed13['category ']
unnamed13.drop(['category ', 'Unnamed: 16', 'Unnamed: 15', 'Unnamed: 14'], inplace = True, axis = 1)


<ipython-input-31-f8e9f8697b0c>:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  unnamed13['name '] = unnamed13['name '] + unnamed13['category ']
C:\Users\blee\Anaconda3\lib\site-packages\pandas\core\frame.py:4163: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  return super().drop(


In [32]:
unnamed13.columns = column_names
unnamed13

ID                                               name  \
1454    1008705746  Zephyra´s new full length 'As The World Collap...   
1563    1009317190            French Cuisine A Traditional Experience   
1794    1010871699  The Beginners Guide to being Unsuicidal the on...   
1931    1011687764        Best OnLine Classifieds Ever / No More Spam   
2420     101453314  Social Media Ruined My Life A Short Film from ...   
...            ...                                                ...   
321945   989007729   THROUGH MY EYES MY LIFE IN THE MISSISSIPPI DELTA   
322162   990511774  Daniel Hresko's new CD is (almost) ready so ge...   
322204   990746749              Feet on the Ground Head in the Clouds   
323138   996542939  'WANDER' - Apocalyptic Short Film Post Production   
323495    99851211  Art/sustainability/spirituality a bike quest i...   

              category main_category currency          deadline   goal  \
1454             Metal         Music      SEK     2/2/2016 0:56  15000   
1563         Cookbooks          Food      USD     9/8/2014 0:46  13730   
1794           Theater       Theater      USD  12/31/2011 23:25   5000   
1931               Web    Technology      USD   9/20/2014 19:56   6300   
2420            Shorts  Film & Video      USD   3/14/2013 20:11   3000   
...                ...           ...      ...               ...    ...   
321945  Narrative Film  Film & Video      USD     4/5/2012 2:34   3500   
322162      Indie Rock         Music      USD     9/6/2011 5:59    400   
322204    Film & Video  Film & Video      USD   4/17/2013 16:00  35000   
323138          Shorts  Film & Video      GBP   4/26/2015 17:12   1500   
323495     Photography   Photography      USD    3/5/2013 21:00   2500   

                launched pledged       state backers country  usd_pledged  
1454       1/3/2016 0:56    4262      failed      14      SE  504.9476528  
1563       8/9/2014 3:16    3984      failed      46      US         3984  
1794    11/21/2011 23:25     525      failed      10      US          525  
1931     8/21/2014 19:56       0      failed       0      US            0  
2420     2/25/2013 21:11    3035  successful      42      US         3035  
...                  ...     ...         ...     ...     ...          ...  
321945     3/6/2012 2:34      10      failed       1      US           10  
322162    8/6/2011 15:36      61      failed       4      US           61  
322204   3/18/2013 15:00     179      failed      10      US          179  
323138   2/25/2015 17:12    2501  successful      50      GB   3861.84447  
323495     2/5/2013 8:09    2634  successful      34      US         2634  

[613 rows x 13 columns]

In [33]:
data2016.drop(unnamed13.index, inplace = True)

In [34]:
data2016.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 323125 entries, 0 to 323749
Data columns (total 17 columns):
 #   Column          Non-Null Count   Dtype  
---  ------          --------------   -----  
 0   ID              323125 non-null  int64  
 1   name            323121 non-null  object 
 2   category        323125 non-null  object 
 3   main_category   323125 non-null  object 
 4   currency        323125 non-null  object 
 5   deadline        323125 non-null  object 
 6   goal            323125 non-null  object 
 7   launched        323125 non-null  object 
 8   pledged         323125 non-null  object 
 9   state           323125 non-null  object 
 10  backers         323125 non-null  object 
 11  country         323125 non-null  object 
 12  usd pledged     319335 non-null  object 
 13  Unnamed: 13     0 non-null       object 
 14  Unnamed: 14     0 non-null       object 
 15  Unnamed: 15     0 non-null       object 
 16  Unnamed: 16     0 non-null       float64
dtypes: float64

In [35]:
data2016.drop(['Unnamed: 13', 'Unnamed: 14', 'Unnamed: 15', 'Unnamed: 16'], axis = 1, inplace = True)

In [36]:
data2016.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 323125 entries, 0 to 323749
Data columns (total 13 columns):
 #   Column          Non-Null Count   Dtype 
---  ------          --------------   ----- 
 0   ID              323125 non-null  int64 
 1   name            323121 non-null  object
 2   category        323125 non-null  object
 3   main_category   323125 non-null  object
 4   currency        323125 non-null  object
 5   deadline        323125 non-null  object
 6   goal            323125 non-null  object
 7   launched        323125 non-null  object
 8   pledged         323125 non-null  object
 9   state           323125 non-null  object
 10  backers         323125 non-null  object
 11  country         323125 non-null  object
 12  usd pledged     319335 non-null  object
dtypes: int64(1), object(12)
memory usage: 34.5+ MB


In [37]:
data2016.columns = column_names

In [38]:
data2016 = data2016.append([unnamed13, unnamed14, unnamed15, unnamed16])

In [39]:
data2016.reset_index(drop = True)

ID                                               name  \
0       1000002330                    The Songs of Adelaide & Abullah   
1       1000004038                                     Where is Hank?   
2       1000007540  ToshiCapital Rekordz Needs Help to Complete Album   
3       1000011046  Community Film Project: The Art of Neighborhoo...   
4       1000014025                               Monarch Espresso Bar   
...            ...                                                ...   
323745   890269949  Community Clothing. Make Clothes Create Jobs R...   
323746  1618382802  Druid Hill Park Passport: Discover Enjoy Learn...   
323747  1864561559        T-shirt dress shirts pants clothing company   
323748   642928449  FRANCIS - The Pope Buenos Aires soccer tango a...   
323749   677103185    SixSixSeven  Angels  Demons  Religion  Esoteric   

              category main_category currency         deadline   goal  \
0               Poetry    Publishing      GBP  10/9/2015 11:36   1000   
1       Narrative Film  Film & Video      USD   2/26/2013 0:20  45000   
2                Music         Music      USD   4/16/2012 4:24   5000   
3         Film & Video  Film & Video      USD   8/29/2015 1:00  19500   
4          Restaurants          Food      USD   4/1/2016 13:38  50000   
...                ...           ...      ...              ...    ...   
323745         Fashion       Fashion      GBP  3/16/2016 16:24  75000   
323746      Publishing    Publishing      USD   7/28/2012 1:30   9500   
323747         Apparel       Fashion      USD   8/2/2015 18:23  20000   
323748    Film & Video  Film & Video      USD   3/2/2014 23:34  75000   
323749  Graphic Novels        Comics      USD  10/10/2015 1:00    750   

               launched  pledged       state backers country usd_pledged  
0       8/11/2015 12:12        0      failed       0      GB           0  
1        1/12/2013 0:20      220      failed       3      US         220  
2        3/17/2012 3:24        1      failed       1      US           1  
3         7/4/2015 8:35     1283    canceled      14      US        1283  
4       2/26/2016 13:38    52375  successful     224      US       52375  
...                 ...      ...         ...     ...     ...         ...  
323745  2/15/2016 17:24  88619.2  successful    1020      GB      128535  
323746   6/6/2012 23:54     9854  successful     208      US        9854  
323747   7/3/2015 18:23        0      failed       0      US           0  
323748    2/5/2014 2:06        0      failed       0      US           0  
323749  9/10/2015 18:15       25      failed       1      US          25  

[323750 rows x 13 columns]

In [40]:
data2016.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 323750 entries, 0 to 269970
Data columns (total 13 columns):
 #   Column         Non-Null Count   Dtype 
---  ------         --------------   ----- 
 0   ID             323750 non-null  int64 
 1   name           323741 non-null  object
 2   category       323750 non-null  object
 3   main_category  323750 non-null  object
 4   currency       323750 non-null  object
 5   deadline       323750 non-null  object
 6   goal           323750 non-null  object
 7   launched       323750 non-null  object
 8   pledged        323750 non-null  object
 9   state          323750 non-null  object
 10  backers        323750 non-null  object
 11  country        323750 non-null  object
 12  usd_pledged    319960 non-null  object
dtypes: int64(1), object(12)
memory usage: 34.6+ MB


Notice that the number of entries (323750) is now the same as what we started with, but the data is cleaned, so that the columns are aligned. There are still some null values in the 'name', and 'usd pledged' column. We'll spend some time looking at those now

In [41]:
no_name = data2016[data2016['name'].isnull() == True]
no_name

ID name            category main_category currency  \
142830  1848699072  NaN      Narrative Film  Film & Video      USD   
262901   634871725  NaN         Video Games         Games      GBP   
265251   648853978  NaN      Product Design        Design      USD   
289847   796533179  NaN            Painting           Art      USD   
36671   1218074363  NaN           Webseries  Film & Video      GBP   
41069    124438738  NaN  Interactive Design        Design      USD   
63544   1378236004  NaN             Apparel       Fashion      USD   
96753   1574873938  NaN                 Art           Art      USD   
269930   676846639  NaN              Design        Design      USD   

               deadline    goal          launched   pledged       state  \
142830  2/29/2012 15:04  200000    1/1/2012 12:35       100      failed   
262901   1/6/2013 23:00    2000  12/19/2012 23:57       196      failed   
265251   7/18/2016 5:01    2500    6/18/2016 5:01         0   suspended   
289847   12/5/2011 5:59   35000   11/6/2011 23:55       220      failed   
36671   5/16/2014 22:25    2000   3/27/2014 21:25         5      failed   
41069   3/25/2015 20:51    9750   2/23/2015 21:51  10890.45  successful   
63544    5/28/2016 0:00    1800   4/29/2016 21:04      2308  successful   
96753    4/3/2014 17:52    7500    3/4/2014 17:52       100      failed   
269930   4/4/2011 20:08   50000    3/5/2011 20:08      1665    canceled   

       backers country  usd_pledged  
142830       1      US          100  
262901      12      GB  317.7284362  
265251       0      US            0  
289847       5      US          220  
36671        1      GB   8.26779725  
41069      107      US     10890.45  
63544      119      US         2308  
96753        1      US          100  
269930      10      US         1665

There really isn't a way for us to guess what the name of a project would be or how we can fill this. But, since our machine learning model will most likely not use the 'name' column to as a predictor, we are safe to leave these as NaN, since they have values in all the other columns we would like to draw conclusions from. 

In [42]:
data2016['usd_pledged'].isna().sum()

3790

In [43]:
usd_pledged_isnull = data2016[data2016['usd_pledged'].isnull()]
usd_pledged_isnull

ID                                               name  \
150     1000694855                      STREETFIGHTERZ WHEELIE MURICA   
287      100149523                        Duncan Woods - Chameleon EP   
549     1003023003          The Making of Ashley Kelley's Debut Album   
561     1003130892                       Butter Side Down Debut Album   
650     1003629045                            Chase Goehring debut EP   
...            ...                                                ...   
323391   997971307  EQUUS International Film Festival®, all-equine...   
323456   998319149                            Emily 2050 - Short Film   
323561     9988744                             Matthew Stephens Music   
323683   999610349     Lady Vendredi: Afrofuturist concept 12 inch EP   
323685   999621278                     Jeune auteur a besoin de vous!   

            category main_category currency          deadline  goal  \
150     Film & Video  Film & Video      USD    9/20/2014 6:59  6500   
287            Music         Music      AUD   8/25/2015 23:00  4500   
549            Music         Music      USD    4/9/2015 21:06  3500   
561            Music         Music      USD  11/26/2015 10:59  6000   
650            Music         Music      USD    3/21/2016 6:00  3000   
...              ...           ...      ...               ...   ...   
323391  Film & Video  Film & Video      USD     2/3/2015 4:19  7500   
323456  Film & Video  Film & Video      CAD    5/23/2014 0:30  3000   
323561         Music         Music      USD    2/5/2016 23:00  5000   
323683         Music         Music      GBP  10/19/2015 22:33  2000   
323685    Publishing    Publishing      EUR  11/20/2015 21:28  1600   

                launched  pledged      state backers country usd_pledged  
150       8/6/2014 21:28      555  undefined       0    N,"0         NaN  
287       8/4/2015 12:05     4767  undefined       0    N,"0         NaN  
549      3/10/2015 20:06     3576  undefined       0    N,"0         NaN  
561      11/2/2015 22:09   7007.8  undefined       0    N,"0         NaN  
650       2/23/2016 3:09  3660.38  undefined       0    N,"0         NaN  
...                  ...      ...        ...     ...     ...         ...  
323391    12/5/2014 4:19       10  undefined       0    N,"0         NaN  
323456     4/8/2014 0:30     3102  undefined       0    N,"0         NaN  
323561    1/6/2016 21:59      235  undefined       0    N,"0         NaN  
323683   9/21/2015 22:33     2125  undefined       0    N,"0         NaN  
323685  10/21/2015 21:28        0  undefined       0    N,"0         NaN  

[3790 rows x 13 columns]

3790 is a lot of entries, it doesn't make sense to drop these, but at first glance, I noticed was that these all have "undefined" as the state of the project. Let's make sure that is the case. 

In [44]:
usd_pledged_isnull[usd_pledged_isnull['state'] != 'undefined']

ID                                               name  \
2102    1012744036          An Oratorio for our Time - Last Stop Cafe   
3736    1022254171         Help OrphanSporks Record Our Fourth Album!   
4202    1024989802                   Jackson Ruby: The Cassette Album   
4598    1027275369        Help Parker Brown make his first solo album   
4799    1028691308          Help BETHANY record a NEW single in Nash!   
...            ...                                                ...   
317438   961951460  Jahque Inc. is merging with Reesi's Cakes 'N T...   
317530   962477474                              Spoken Fruit Magazine   
318376   967557880             Help Danelle Finish Her First EP Album   
320937   982919439                                              Faith   
323352   997793413         Whisper Violet - Honors Senior Thesis Film   

            category main_category currency         deadline   goal  \
2102           Music         Music      USD  5/26/2016 19:46   5000   
3736           Music         Music      USD   5/14/2016 5:59   2500   
4202           Music         Music      USD  5/11/2016 19:29   5000   
4598           Music         Music      USD  5/21/2016 16:14   3800   
4799           Music         Music      USD   5/8/2016 22:37   3500   
...              ...           ...      ...              ...    ...   
317438        Events          Food      USD    5/5/2016 6:01  25000   
317530         Print    Journalism      EUR  4/28/2016 16:02  15000   
318376         Music         Music      USD  4/28/2016 23:12   3500   
320937  Film & Video  Film & Video      USD    5/6/2016 4:17   9000   
323352  Film & Video  Film & Video      USD   5/14/2016 1:27   2000   

               launched  pledged       state backers country usd_pledged  
2102    4/26/2016 19:46     5170  successful       0    N,"0         NaN  
3736     4/11/2016 7:20     1375    canceled       0    N,"0         NaN  
4202    4/11/2016 19:29     5296  successful       0    N,"0         NaN  
4598    4/21/2016 16:14     5077  successful       0    N,"0         NaN  
4799     4/4/2016 23:32  3502.13  successful       0    N,"0         NaN  
...                 ...      ...         ...     ...     ...         ...  
317438    4/5/2016 6:01        0        live       0    N,"0         NaN  
317530  3/29/2016 16:02     1933      failed       0    N,"0         NaN  
318376   4/5/2016 18:56     3686  successful       0    N,"0         NaN  
320937    3/7/2016 4:17        2      failed       0    N,"0         NaN  
323352   4/14/2016 1:27     2300  successful       0    N,"0         NaN  

[235 rows x 13 columns]

It looks like there are other values for "state" besides undefined. But What about the "backers" column. They all seem to be 0 at first glance. Per the kickstarter website, backers are the number of people who "pledge" money, so theoretically, there shouldn't be any pledges if there are no backers. Let's confirm if all of those listed in usd_pledged with NaN values have 0 backers


In [45]:
usd_pledged_isnull[usd_pledged_isnull['backers'] != '0']


Empty DataFrame
Columns: [ID, name, category, main_category, currency, deadline, goal, launched, pledged, state, backers, country, usd_pledged]
Index: []

In [46]:
usd_pledged_isnull['backers'].nunique()

1

There is only 1 unique value under backers, we can assume that the unique value is 0. Since it doesn't make sense that a project can have pledges, or be successful if there are no backers, these seem like data collection or entry errors. We do not have access to find this missing data to course correct. We also want to use the number of backers, pledged, and usd_pledged coloumns as factors in predicting the state, it makes sense to drop these rows that have missing or incomplete data. 

In [47]:
data2016.drop(usd_pledged_isnull.index, inplace = True)

In [48]:
data2016.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 319960 entries, 0 to 269970
Data columns (total 13 columns):
 #   Column         Non-Null Count   Dtype 
---  ------         --------------   ----- 
 0   ID             319960 non-null  int64 
 1   name           319951 non-null  object
 2   category       319960 non-null  object
 3   main_category  319960 non-null  object
 4   currency       319960 non-null  object
 5   deadline       319960 non-null  object
 6   goal           319960 non-null  object
 7   launched       319960 non-null  object
 8   pledged        319960 non-null  object
 9   state          319960 non-null  object
 10  backers        319960 non-null  object
 11  country        319960 non-null  object
 12  usd_pledged    319960 non-null  object
dtypes: int64(1), object(12)
memory usage: 34.2+ MB


We dropped 3790 rows from the orignal of 323750, so the total number of rows now should be 319960 which matches what we see above. 

Now our task is to make sure the datatypes are correct. 
category, main_category, currency, and country should be changed to category data types
deadline and launched should be date/time
goal, pledged, backers, and usd_pledged should all be type int. 

In [49]:
data2016['goal'].map(type)

0         <class 'str'>
1         <class 'str'>
2         <class 'str'>
3         <class 'str'>
4         <class 'str'>
              ...      
305503    <class 'str'>
104120    <class 'str'>
145502    <class 'str'>
264243    <class 'str'>
269970    <class 'str'>
Name: goal, Length: 319960, dtype: object

In [50]:
temp = data2016[data2016['goal'].map(type) != str]

In [51]:
temp['goal'].map(type)

Series([], Name: goal, dtype: object)

Since all the rows are currently listed as type object, its a mix of str and numerical values. We aren't able to use the map function to pull these out. 

In [52]:
data2016['goal'] = data2016['goal'].apply(pd.to_numeric, errors='coerce')


In [53]:
data2016.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 319960 entries, 0 to 269970
Data columns (total 13 columns):
 #   Column         Non-Null Count   Dtype  
---  ------         --------------   -----  
 0   ID             319960 non-null  int64  
 1   name           319951 non-null  object 
 2   category       319960 non-null  object 
 3   main_category  319960 non-null  object 
 4   currency       319960 non-null  object 
 5   deadline       319960 non-null  object 
 6   goal           319953 non-null  float64
 7   launched       319960 non-null  object 
 8   pledged        319960 non-null  object 
 9   state          319960 non-null  object 
 10  backers        319960 non-null  object 
 11  country        319960 non-null  object 
 12  usd_pledged    319960 non-null  object 
dtypes: float64(1), int64(1), object(11)
memory usage: 34.2+ MB


since there are only 7 rows that seem to be missaligned, where the 'coerce' filled it with NaN, let's drop those instead of trying to realign them. 

In [54]:
goal_nan = data2016[data2016['goal'].isnull()]
goal_nan

ID                               name  \
64484   1383759393            CD: Heartsong Harmonics   
85619   1508366697                 Everyday Beautiful   
98614   1585608231              Hana Zara's new album   
123506  1733391784          Celebrating Joni Mitchell   
159427   194816108                        Legal Highs   
163067  1969863991               Puppy Training Steps   
177636  2057841246  Adventure to Peru's Sacred Valley   

                                   category main_category      currency  \
64484    Marc & Mary sing together at last!         Music         Music   
85619       a day in the life of a champion  Film & Video  Film & Video   
98614                      The North (2015)         Music         Music   
123506       Songs by & about her + Stories         Music         Music   
159427                   The Sobering Truth  Film & Video  Film & Video   
163067                      Right and Wrong  Film & Video  Film & Video   
177636                Recording Soundtracks         Music         Music   

       deadline  goal launched          pledged state    backers country  \
64484       USD   NaN     7085  9/14/2015 23:55  7470  undefined       0   
85619       USD   NaN     2000   2/25/2015 6:52     0  undefined       0   
98614       USD   NaN     3000   1/17/2015 2:04  3280  undefined       0   
123506      CAD   NaN     5000   9/11/2015 0:47  6323  undefined       0   
159427      GBP   NaN     5000  10/16/2014 3:00     0  undefined       0   
163067      USD   NaN     5000  3/25/2015 21:07     0  undefined       0   
177636      USD   NaN     2200   3/2/2015 22:14  2503  undefined       0   

       usd_pledged  
64484         N,"0  
85619         N,"0  
98614         N,"0  
123506        N,"0  
159427        N,"0  
163067        N,"0  
177636        N,"0

In [55]:
data2016.drop(goal_nan.index, inplace = True)

In [56]:
data2016.reset_index(drop = True)

ID                                               name  \
0       1000002330                    The Songs of Adelaide & Abullah   
1       1000004038                                     Where is Hank?   
2       1000007540  ToshiCapital Rekordz Needs Help to Complete Album   
3       1000011046  Community Film Project: The Art of Neighborhoo...   
4       1000014025                               Monarch Espresso Bar   
...            ...                                                ...   
319948   890269949  Community Clothing. Make Clothes Create Jobs R...   
319949  1618382802  Druid Hill Park Passport: Discover Enjoy Learn...   
319950  1864561559        T-shirt dress shirts pants clothing company   
319951   642928449  FRANCIS - The Pope Buenos Aires soccer tango a...   
319952   677103185    SixSixSeven  Angels  Demons  Religion  Esoteric   

              category main_category currency         deadline     goal  \
0               Poetry    Publishing      GBP  10/9/2015 11:36   1000.0   
1       Narrative Film  Film & Video      USD   2/26/2013 0:20  45000.0   
2                Music         Music      USD   4/16/2012 4:24   5000.0   
3         Film & Video  Film & Video      USD   8/29/2015 1:00  19500.0   
4          Restaurants          Food      USD   4/1/2016 13:38  50000.0   
...                ...           ...      ...              ...      ...   
319948         Fashion       Fashion      GBP  3/16/2016 16:24  75000.0   
319949      Publishing    Publishing      USD   7/28/2012 1:30   9500.0   
319950         Apparel       Fashion      USD   8/2/2015 18:23  20000.0   
319951    Film & Video  Film & Video      USD   3/2/2014 23:34  75000.0   
319952  Graphic Novels        Comics      USD  10/10/2015 1:00    750.0   

               launched  pledged       state backers country usd_pledged  
0       8/11/2015 12:12        0      failed       0      GB           0  
1        1/12/2013 0:20      220      failed       3      US         220  
2        3/17/2012 3:24        1      failed       1      US           1  
3         7/4/2015 8:35     1283    canceled      14      US        1283  
4       2/26/2016 13:38    52375  successful     224      US       52375  
...                 ...      ...         ...     ...     ...         ...  
319948  2/15/2016 17:24  88619.2  successful    1020      GB      128535  
319949   6/6/2012 23:54     9854  successful     208      US        9854  
319950   7/3/2015 18:23        0      failed       0      US           0  
319951    2/5/2014 2:06        0      failed       0      US           0  
319952  9/10/2015 18:15       25      failed       1      US          25  

[319953 rows x 13 columns]

In [58]:
data2016 = data2016.astype({'name':'str', 
                            'category':'str', 
                            'main_category':'category', 
                            'currency':'category',
                            'deadline':'datetime64', 
                            'launched':'datetime64', 
                            'pledged':'float',
                            'state' : 'category',
                            'backers' : 'int64', 
                            'country' : 'category',
                            'usd_pledged':'float'})
                            
                            

In [59]:
data2016.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 319953 entries, 0 to 269970
Data columns (total 13 columns):
 #   Column         Non-Null Count   Dtype         
---  ------         --------------   -----         
 0   ID             319953 non-null  int64         
 1   name           319953 non-null  object        
 2   category       319953 non-null  object        
 3   main_category  319953 non-null  category      
 4   currency       319953 non-null  category      
 5   deadline       319953 non-null  datetime64[ns]
 6   goal           319953 non-null  float64       
 7   launched       319953 non-null  datetime64[ns]
 8   pledged        319953 non-null  float64       
 9   state          319953 non-null  category      
 10  backers        319953 non-null  int64         
 11  country        319953 non-null  category      
 12  usd_pledged    319953 non-null  float64       
dtypes: category(4), datetime64[ns](2), float64(3), int64(2), object(2)
memory usage: 25.6+ MB


perfect, now that the dtypes are assinged, we are able to start analyzing further. 

In [60]:
data2016.describe()

ID          goal       pledged        backers   usd_pledged
count  3.199530e+05  3.199530e+05  3.199530e+05  319953.000000  3.199530e+05
mean   1.074852e+09  4.761360e+04  8.754013e+03     102.719706  7.838580e+03
std    6.193962e+08  1.145230e+06  8.989582e+04     939.483840  8.460305e+04
min    5.971000e+03  1.000000e-02  0.000000e+00       0.000000  0.000000e+00
25%    5.374037e+08  2.000000e+03  3.000000e+01       2.000000  2.500000e+01
50%    1.075858e+09  5.000000e+03  6.100000e+02      12.000000  5.350000e+02
75%    1.611070e+09  1.500000e+04  3.934000e+03      55.000000  3.571000e+03
max    2.147476e+09  1.000000e+08  2.033899e+07  219382.000000  2.033899e+07

Summary

1. Columns and values were misaligned. I created temporary dataframes, adjusted the columns and values to match, then merged back into main dataframe, deleted the duplicates. 

2. found missing data in the names column, decided that is not important and will not be used as part of predictive model, but since the other columns have values, we are keeping it

3. Found rows with inconsistent data, you cannot have pledge values with a NaN or 0 backer value. Decided to delete those rows since we are unable to go back and verify this data

4. Changed the column datatypes to what matches best, deleted another 7 rows since they were still misaligned. 

In [61]:
data2016.to_csv('clean_data2016.csv')